plugins/catalog-graphql/src/service/client.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity, EntityMeta } from '@backstage/catalog-model';
+import fetch from 'node-fetch';
+import { JsonObject } from '@backstage/config';
+
+export interface ReaderEntityMeta extends EntityMeta {
+  uid: string;
+  etag: string;
+  generation: number;
+  namespace: string;
+  annotations: Record<string, string>;
+  labels: Record<string, string>;
```

Comment:
are these three really non-nullable from a reader's POV? seems to me that that would require some more work on our part to ensure that they are present

Reply:
We already do here:
https://github.com/spotify/backstage/blob/3f601b60558cf6bd652c66d9729f8f5f06935925/plugins/catalog-graphql/src/graphql/module.ts#L76-L77